dashboard/src/reducers/auth.ts

```diff
@@ -6,25 +6,33 @@ import { AuthAction } from "../actions/auth";
 export interface IAuthState {
   authenticated: boolean;
   authenticating: boolean;
+  oidcAuthenticated: boolean;
   authenticationError?: string;
 }
 
 const initialState: IAuthState = {
   authenticated: !(localStorage.getItem("kubeapps_auth_token") === null),
   authenticating: false,
+  oidcAuthenticated: localStorage.getItem("kubeapps_auth_token_oidc") === "true",
 };
 
 const authReducer = (state: IAuthState = initialState, action: AuthAction): IAuthState => {
   switch (action.type) {
     case getType(actions.auth.setAuthenticated):
-      return { ...state, authenticated: action.payload, authenticating: false };
+      return {
+        ...state,
+        authenticated: action.payload.authenticated,
+        oidcAuthenticated: !!action.payload.oidc,
```

Comment:
yes it's a boolean but it's optional, so that's why I needed the conversion. In any case there is not a requirement for this to be optional so we can avoid this.